cpmpy/transformations/linearize.py

```diff
@@ -179,6 +199,18 @@ def linearize_constraint(lst_of_expr, supported={"sum","wsum"}, reified=False):
             [cpm_expr] = canonical_comparison([cpm_expr])  # just transforms the constraint, not introducing new ones
             lhs, rhs = cpm_expr.args
 
+            # check trivially true/false (not allowed by PySAT Card/PB)
+            if cpm_expr.name in ('<', '<=', '>', '>=') and is_num(rhs):
```

Comment:
What if i post a constraint `x + y + z == 10`, this is trivially UNSAT but not catched here?

Reply:
In the future, we can actually move these checks into the `get_bounds` function on Comparisons (and Boolean operators)

Reply:
I've added a test for this, it seems to be ok.

I've realized that instead of guessing when PySAT will raise an exception for trivial unsatisfiable constraints, we should just catch this error. However, there are two issues from preventing this approach at the moment, for which I created a ticket for it https://github.com/pysathq/pysat/issues/194). Essentially, PySAT has a case where it is incorrect about trivial unsat, and it throws a generic ValueError (which is dangerous to catch). It also seems they do support negative weights. So the integration can be much simplified.

The question is whether we fix the current clunky approach or whether we wait for a PySAT release (potentially with our own PR).

Reply:
If its correct now, you can merge it now.

If it becomes easier later, we can make it easier later : )